UM-USElab/gradecraft-development
# Learning objectives should have at least two levels

### Status
**READY**

### Dev note:
up     20190702175922  Add default level to learning objective level
switching between branches turn down migration with:

`rake db:migrate:down VERSION=20190702175922`

### Description
* A learning objective should have at least two learning objective levels. Currently it is possible to have a learning objective with no learning objective levels or one learning objective level.
* Now, creating a learning objective creates two learning objectives that cannot be deleted, one with the minimum proficiency level and one with the maximum proficiency level. Additional learning objective levels can be created and deleted as necessary however every learning objective will have at least two learning objective levels.

### Related PRs
`add-lo-assignment-overview` [#4108]

### Deploy Notes
* A new boolean value called default_level has been added to the LearningObjectiveLevel model for checking whether a learning objective level was created when the learning objective was created or whether the level as created by the user.

### Migrations
YES

### Steps to Test or Reproduce
1. As an instructor, create a learning objective.
2. The learning objective will have two levels one with the minimum proficiency level set and one with the maximum proficiency level set. It should not be possible to delete these two levels, but it is possible to add and delete more learning objective levels if required by the instructor.

### Impacted Areas in Application
* Creating learning objective (i.e. models/learning_objective.rb,  controllers/api/learning_objectives/objectives_controller.rb)
* Learning objective edit page

======================
Closes #4114
